test: smp :add testcase

yingmingx opened the pull request:
This is an integration testcase for smp. It tests
the situation when smp is configed. Fatal can be invoked on
different core and system workq can also be run on different
core.


Signed-off-by: Ying ming <mingx.ying@intel.com>

nashif commented:
Can this be folded into an existing SMP test please?

andyross reviewed (commented):
Why is this a separate test?  Can't you just enable SMP in the existing fatal test case (or rather: just leave it unset and leave the platform choice in place) and add these as two new cases?  Is that broken in some way?  Seems like that's something we should fix instead?

Commit:
test: smp :add testcase

This is an integration testcase for smp. It tests
the situation when smp is configed. Fatal can be invoked on
different core and system workq can also be run on different
core.

Signed-off-by: Ying ming <mingx.ying@intel.com>

yingmingx commented:
I have put this testcase into tests/kernel/smp because we classify testcases in this directory are integration test and  testcases in tests/kernel/fatal/exception are module test.
